fastplotlib/layouts.py

```diff
@@ -111,9 +119,17 @@ def __init__(
                 renderer=renderer
             )
 
+            self.indexers[i, j] = GridPlotIndexer((i,j))
+
         self._animate_funcs: List[callable] = list()
         self._current_iter = None
 
+    def __getitem__(self, index: tuple):
+        return self.indexers[index[0], index[1]].name
```

Comment:
you don't return the name, just the instance of `GridPlotIndexer`

Reply:
Also, we want to allow the `index` argument to be either a `tuple[int, int]` or `str` with the name. It then iterates through all `self.indexers` until it finds a match.